crates/agentgateway/src/mcp/relay/pool.rs

```diff
@@ -452,18 +468,34 @@ pub struct ClientWrapper {
 	backend: Arc<SimpleBackend>,
 	client: PolicyClient,
 	policies: BackendPolicies,
+	headers: http::HeaderMap,
+}
+
+impl ClientWrapper {
+	pub fn insert_headers(&self, req: &mut crate::http::Request) {
+		for (k, v) in &self.headers {
+			if !req.headers().contains_key(k) {
+				req.headers_mut().insert(k.clone(), v.clone());
+			}
+		}
+	}
 }
 
 impl ClientWrapper {
 	pub fn new_with_client(
 		backend: SimpleBackend,
 		client: PolicyClient,
 		policies: BackendPolicies,
+		mut headers: HeaderMap,
 	) -> Self {
+		// Remove headers we do not want to propagate to the backend
+		headers.remove(http::header::CONTENT_ENCODING);
```

Comment:
Might be worth documenting this somewhere just in case, outside of just code I mean.